hazelcast/src/main/java/com/hazelcast/partition/impl/ResetReplicaVersionOperation.java

```diff
@@ -34,6 +36,10 @@ public void run() throws Exception {
         int partitionId = getPartitionId();
         InternalPartitionService partitionService = getService();
         long[] versions = partitionService.getPartitionReplicaVersions(partitionId);
+        // InternalPartitionService.getPartitionReplicaVersions() returns internal
+        // version array, we need to clone it here
+        versions = Arrays.copyOf(versions, InternalPartition.MAX_BACKUP_COUNT);
```

Comment:
Thanks to @metanet, who found this issue and fixed in his repo:
https://github.com/metanet/hazelcast/commit/7a1f882c961bb56c7f10c6d20a5e2acf446944e4